portiaui/app/models/base.js

```diff
@@ -3,8 +3,110 @@ import DS from 'ember-data';
 import EmptyObject from "ember-data/-private/system/empty-object";
 const { get } = Ember;
 
+let currentActionModel = null;
+const modelActionQueue = [];
+
+function runActions() {
+    if (currentActionModel && currentActionModel.get('isSaving')) {
+        // current action has not finished
+        return;
+    }
+    currentActionModel = null;
+    while (modelActionQueue.length) {
+        const nextModel = modelActionQueue.shift();
+        if (nextModel.get('isSaving')) {
+            // model has started saving again, wait for the next didCommit event
+            return;
+        }
+        if (nextModel.get('isDeleted') && !nextModel.get('hasDirtyAttributes')) {
+            // model has been deleted
+            continue;
+        }
+        if (nextModel.get('pendingDelete')) {
+            nextModel.deleteRecord();
+            nextModel.set('pendingDelete', false);
```

Comment:
After deleting a record don't we want to `continue`? Is there a case when a record has `pendingDelete === true` and `pendingSave === true`? With `continue` it would be clearer if it is the case that deleting a record just continues to the next model in `modelActionQueue`.